phase2/kubeadm/configure-vm-kubeadm-master.sh

```diff
@@ -43,4 +42,25 @@ kubeProxy:
 EOF
 fi
 
+# if exist $KUBEADM_FEATURE_GATES is an simple array with "key1 value1 key2 value2 .."
+# it should be added in the config like:
+# featureGates:
+#   key1: value1
+#   key2: value2
+#   etc ...
+featuresdata=($KUBEADM_FEATURE_GATES)
+featureslength=(${#featuresdata[*]})
+if (( $featureslength % 2 == 0 )) && (( $featureslength > 0 )); then
+  configfeatures="featureGates: "
+  for ((ind=0; ind<$featureslength; ind++, ind++));
+  do
+    configfeatures="$configfeatures
+  ${featuresdata[ind]}: ${featuresdata[ind+1]}"
```

Comment:
same as above comment

Reply:
My problem is the format of the parameter : at the end, in YAML file for kubeadmin, it is expected a multiline value. 
A way to avoid the loop in script and evaluation of expr, 
At the end I need: 

featureGates:
   <key1>: <value1>
   <key2>: <value2>
   ...

the list of key/values is provided by the initial parameter in jobs .. 

I do not like too much to prepare that format from jsonnet or terrarform  - that means pushed this parameter already YAML formatted in the cloud-init parameters (supposing I can provide here a multiline value without crazy encoding).

Best would be to have a tool at startup-script that could do the transformation (a simple function but not bash .. or a linux tool that can do csv-like to yaml).

Let me make some search here.